ansible_collections/arista/avd/plugins/README.md

```diff
@@ -176,6 +176,21 @@ To use this filter:
 !!! note
     This isn't using the same range syntax as EOS for modular or break-out ports. For example, on EOS `et1/1-2/4` gives you `et1/1, et1/2, et1/3, et1/4, et2/1, et2/2, et2/3, et2/4` on a fixed switch, but a different result on a modular switch depending on the module types. In AVD, the same range would be `et1-2/1-4`.
 
+#### bgp_encrypt / bgp_decrypt filters
+
+The `arista.avd.bgp_encrypt` and `arista.avd.bgp_decrypt` filters are used to encrypt or decrypt BGP passwords based on the Neighbor IP or the BGP Peer Group Name as is expected in EOS.
+
+An example usage for `arista.avd.bgp_encrypt` filter it ot use it in conjunction with Ansible Vault to be able to load a password and have it encrypted on the fly by AVD in `eos_designs`.
+
+**example:**
+
+```jinja
+bgp_peer_groups:
+  ipv4_underlay_peers:
+      name: IPv4-UNDERLAY-PEERS
+          password: "{{ bgp_vault_password | bgp_encrypt(key=IPv4-UNDERLAY-PEERS)
```

Comment:
From the kode it looks like you are taking an input dict splitting out "key" and "password". In this example you are passing a string. Also the indentation looks strange here.
I suggest to design the filter with `mypassword | arista.avd.encrypt('bgp', 'IPv4-UNDERLAY-PEERS')` which in python translates to `def encrypt(password, type, key)`

Reply:
Update in be2f78a